lib/menu.ts

```diff
@@ -113,27 +119,60 @@ export class Menu extends Disposable implements IPopupContent {
     );
     this.onDispose(() => domDispose(this.content));
 
+    if (options.startIndex !== undefined) {
+      const elems = Array.from(this.content.children).filter(elem => isSelectable(elem));
+      this._selected.set(elems[options.startIndex]);
+    }
+
     FocusLayer.create(this, this.content);
   }
 
   private _nextIndex(): void {
-    this._selIndex.set((this._selIndex.get() + 1) % this._items.length);
+    const elem = this._selected.get();
+    const content = this.content;
+    if (elem) {
+      const getNext = (_elem: Element) => _elem.nextElementSibling || content.firstElementChild;
+      const next = getNextSelectable(elem, getNext);
+      if (next) { this._selected.set(next); }
+    }
   }
+
   private _prevIndex(): void {
-    const N = this._items.length;
-    this._selIndex.set((Math.max(this._selIndex.get(), 0) + N - 1) % N);
+    const elem = this._selected.get();
+    const content = this.content;
+    if (elem) {
+      const getNext = (_elem: Element) => _elem.previousElementSibling || content.lastElementChild;
+      const next = getNextSelectable(elem, getNext);
+      if (next) { this._selected.set(next); }
+    }
   }
 
   private _onMouseOver(ev: MouseEvent) {
     // Find immediate child of this.content which is an ancestor of ev.target.
-    const child = findAncestorChild(this.content, ev.target as Element);
-    const index = child ? this._items.indexOf(child) : -1;
-    if (index >= 0) {
-      this._selIndex.set(index);
-    }
+    const elem = findAncestorChild(this.content, ev.target as Element);
+    if (elem && isSelectable(elem)) { this._selected.set(elem); }
   }
 }
 
+/**
+ * Given a starting Element and a function to retrieve the next Element, returns the next selectable
+ * Element (based on isSelectable). Returns null if the function to retrieve the next Element returns
+ * null. Always returns the starting element if given by the next Element function to prevent an
+ * infinite loop.
+ */
+function getNextSelectable(startElem: Element, getNext: (elem: Element) => Element|null): Element|null {
+  let next = getNext(startElem);
+  while (next && next !== startElem && !isSelectable(next)) { next = getNext(next); }
+  return next;
+}
+
+/**
+ * Returns a boolean indicating whether the Element is selectable in the menu.
+ */
+function isSelectable(elem: Element): boolean {
+  return elem.hasAttribute('tabIndex') && getComputedStyle(elem).display !== 'none';
```

Comment:
I don't like getComputedStyle() here, I think it's a expensive. How about `elem.offsetHeight > 0` (from https://davidwalsh.name/offsetheight-visibility)